Lib/test/test_functools.py

```diff
@@ -2188,5 +2188,89 @@ def _(arg: typing.Iterable[str]):
         self.assertTrue(str(exc.exception).endswith(msg_suffix))
 
 
+class CachedCostItem:
+    _cost = 1
+
+    def __init__(self):
+        self.lock = py_functools.RLock()
+
+    @py_functools.cached_property
+    def cost(self):
+        """The cost of the item."""
+        with self.lock:
+            self._cost += 1
+        return self._cost
+
+
+class CachedCostItemWait:
+    _cost = 1
+
+    def __init__(self, event):
+        self._cost = 1
+        self.lock = py_functools.RLock()
+        self.event = event
+
+    @py_functools.cached_property
+    def cost(self):
+        self.event.wait(1)
+        with self.lock:
+            self._cost += 1
+        return self._cost
+
+
+class CachedCostItemWithSlots:
+    __slots__ = ('_cost')
+
+    def __init__(self):
+        self._cost = 1
+
+    @py_functools.cached_property
+    def cost(self):
+        """The cost of the item."""
+        return self._cost
```

Comment:
Looks like the method is never called

Reply:
Yes, that’s expected. The point of this class is just to test the error message; cached property on classes with slots is not supported. Would you find it clearer if I used `pass` as the body? I just wrote it the way someone realistically might, to make it clear that the error is not due to a mistake in the implementation of the method.

Reply:
`raise RuntimeError('never executed')` is even more clear.